server_cached_router.go

```diff
@@ -65,22 +84,54 @@ func (r cachedRouter) FindProviders(ctx context.Context, key cid.Cid, limit int)
 }
 
 func (r cachedRouter) FindPeers(ctx context.Context, pid peer.ID, limit int) (iter.ResultIter[*types.PeerRecord], error) {
```

Comment:
Btw, initially I had implemented the cache for `FindPeers` only to remove it later. My reasoning is that if you fail to `FindPeer`, as part of a kad-dht look (which will ask at least 20 peers), you probably don't want to return it from cache.

Do you still see value in returning addresses from cache here? At the very least we could maybe check the last time we connected to them in `withAddrsFromCache`/`GetCachedAddrs`.